src/components/Api.js

```diff
@@ -0,0 +1,108 @@
+export default class Api {
+  constructor({ baseUrl, headers }) {
+    this._baseUrl = baseUrl;
+    this._headers = headers;
+    this._authorization = headers.authorization;
+  }
+
+  _checkServerResponse(res) {
+    if (res.ok) {
+      return res.json();
+    } else {
+      return Promise.reject(`Error: ${res.status}`);
+    }
+  }
+
+  getUsersInfo() {
+    return fetch(`${this._baseUrl}/users/me`, {
+      headers: {
```

Comment:
Needs correction:

1) The method is not a part of the headers
2) You should use the saved headers in all the requests

```js
return fetch(`${this._baseUrl}/users/me`, {
  headers: this._headers
```